Case 20723, Case 20334: Internally, replace all overlays with local entities

SamGondelman opened the pull request:
https://highfidelity.manuscript.com/f/cases/20723/
https://highfidelity.manuscript.com/f/cases/20334/HMD-Login-Zone-tilted-keyboard

Test plan:
- verify that the following work as in master:
    - login screen (desktop and HMD, I think?)
    - tablet (as much as possible.  include: create, home button highlighting, general clicking etc., resizing your avatar)
    - avatar loading spheres (they should blink and change color and disappear)
    - context overlay (floating "i" when you right click on a marketplace item)
    - keyboard focus highlight (when you click on a web entity or overlay, it should have a yellow outline, except for the tablet)
    - create gizmos (in this PR, they will be unlit.  the fact that they were lit was a bug) (should highlight on mouse over, should be clickable and work)
    - 3D keyboard + styluses (all related test rail cases and the test plan from case 20334)
    - hand lasers (rendering, interaction)
    - pal spheres
    - important scripts that make overlays (the shapes app from the marketplace, for example)
- run all tests in Nitpick from this PR: https://github.com/highfidelity/hifi_tests/pull/335

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16764/HighFidelity-Beta-Interface-PR14782-a2083d9.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16782/HighFidelity-Beta-PR14782-bb5c556.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16782/HighFidelity-Beta-Interface-PR14782-bb5c556.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16786/HighFidelity-Beta-PR14782-e15fda3.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16786/HighFidelity-Beta-Interface-PR14782-e15fda3.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16787/HighFidelity-Beta-PR14782-abd7d6b.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16787/HighFidelity-Beta-Interface-PR14782-abd7d6b.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16794/HighFidelity-Beta-PR14782-70b1e17.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16794/HighFidelity-Beta-Interface-PR14782-70b1e17.exe

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16840/HighFidelity-Beta-PR14782-445608e.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16840/HighFidelity-Beta-Interface-PR14782-445608e.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16850/HighFidelity-Beta-PR14782-6ad0b34.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16851/HighFidelity-Beta-PR14782-a8dbe5e.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16851/HighFidelity-Beta-Interface-PR14782-a8dbe5e.exe

SamGondelman commented on interface/src/Application.cpp:
```diff
@@ -3823,28 +3794,6 @@ bool Application::handleKeyEventForFocusedEntityOrOverlay(QEvent* event) {
         }
     }
 
-    if (_keyboardFocusedOverlay.get() != UNKNOWN_OVERLAY_ID) {
```
Before, there was a `_keyboardFocusedOverlay` and a `_keyboardFocusedEntity`, but only one would render at a time anyways.  Now there's just one.

SamGondelman commented on interface/src/devices/DdeFaceTracker.cpp:
```diff
@@ -634,12 +629,8 @@ void DdeFaceTracker::calibrate() {
-        _calibrationBillboard->setLeftMargin(CALIBRATION_BILLBOARD_LEFT_MARGIN);
-        _calibrationBillboard->setFontSize(CALIBRATION_BILLBOARD_FONT_SIZE);
-        _calibrationBillboard->setText(CALIBRATION_INSTRUCTION_MESSAGE);
-        _calibrationBillboard->setAlpha(CALIBRATION_BILLBOARD_ALPHA);
```
These functions don't do anything (see TextOverlay.cpp), so I've removed them.

SamGondelman commented on interface/src/ui/Keyboard.cpp:
```diff
@@ -33,11 +33,6 @@
 #include <RegisteredMetaTypes.h>
```
The changes in this file (and some others) are largely taken directly from @danteruiz's PR: https://github.com/highfidelity/hifi/pull/14685

SamGondelman commented on interface/src/ui/overlays/Base3DOverlay.cpp:
```diff
@@ -1,392 +0,0 @@
-//
```
it feels so good to delete these files

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -50,41 +50,53 @@ Overlays::Overlays() {
+    connect(this, &Overlays::hoverLeaveOverlay, entityScriptingInterface.data(), &EntityScriptingInterface::hoverLeaveEntity);
+    connect(this, &Overlays::mousePressOnOverlay, entityScriptingInterface.data(), &EntityScriptingInterface::mousePressOnEntity);
+    connect(this, &Overlays::mouseMoveOnOverlay, entityScriptingInterface.data(), &EntityScriptingInterface::mouseMoveOnEntity);
+    connect(this, &Overlays::mouseReleaseOnOverlay, entityScriptingInterface.data(), &EntityScriptingInterface::mouseReleaseOnEntity);
```
When you click on an overlay, it generates an overlay event and an entity event.  This makes things like web overlays work without duplicating the code in this file, and also it makes sense because they're local entities.

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -144,490 +156,843 @@ void Overlays::enable() {
 }
 
-OverlayID Overlays::addOverlay(const QString& type, const QVariant& properties) {
+QString Overlays::entityToOverlayType(const QString& type) {
```
The following code is for converting between overlay and entity properties, which of course were slightly different.

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -144,490 +156,843 @@ void Overlays::enable() {
+    if (type == "rectangle3d") {
+        propertyMap["shape"] = "Quad";
+    }
+    return DependencyManager::get<EntityScriptingInterface>()->addEntityInternal(convertOverlayToEntityProperties(propertyMap, entityType, true), entity::HostType::LOCAL);
```
Most of the functions in this class now follow this pattern: first, check if we're dealing with a 2D overlay.  If not, just call the equivalent function in the EntityScriptingInterface.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16852/HighFidelity-Beta-PR14782-9aae9d1.apk

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -813,36 +1142,63 @@ static PointerEvent::Button toPointerButton(const QMouseEvent& event) {
     PerformanceTimer perfTimer("Overlays::mousePressEvent");
 
     PickRay ray = qApp->computePickRay(event->x(), event->y());
-    RayToOverlayIntersectionResult rayPickResult = findRayIntersectionVector(ray, true, QVector<OverlayID>(),
```
Overlays use a Pick internally now 💃

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -1020,69 +1285,689 @@ bool Overlays::mouseMoveEvent(QMouseEvent* event) {
+    if (entityTree) {
+        unsigned int searchFilter = PickFilter::getBitMask(PickFilter::FlagBit::LOCAL_ENTITIES);
+        entityTree->withReadLock([&] {
+            entityTree->evalEntitiesInSphere(center, radius, PickFilter(searchFilter), result);
```
This is also so much faster than the old code!

SamGondelman commented on interface/src/ui/overlays/Overlays.cpp:
```diff
@@ -1020,69 +1285,689 @@ bool Overlays::mouseMoveEvent(QMouseEvent* event) {
-
     return result;
 }
+
```
Below are ALL of the different OverlayProperties, for all subclasses.

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16852/HighFidelity-Beta-PR14782-9aae9d1.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16852/HighFidelity-Beta-Interface-PR14782-9aae9d1.dmg

SamGondelman commented on libraries/render-utils/src/RenderForwardTask.cpp:
```diff
@@ -135,10 +132,14 @@ void RenderForwardTask::build(JobModel& task, const render::Varying& input, rend
+    const auto hudOpaquesInputs = DrawLayered3D::Inputs(hudOpaque, lightingModel, nullJitter).asVarying();
+    const auto hudTransparentsInputs = DrawLayered3D::Inputs(hudTransparent, lightingModel, nullJitter).asVarying();
+    task.addJob<DrawLayered3D>("DrawHUDOpaque", hudOpaquesInputs, true);
+    task.addJob<DrawLayered3D>("DrawHUDTransparent", hudTransparentsInputs, false);
```
Enable HUD layer drawing on the forward path

SamGondelman commented on libraries/render/src/render/RenderFetchCullSortTask.h:
```diff
@@ -23,8 +23,6 @@ class RenderFetchCullSortTask {
         LIGHT,
         META,
-        OVERLAY_OPAQUE_SHAPE,
-        OVERLAY_TRANSPARENT_SHAPE,
```
Completely unused 🤷‍♂️

SamGondelman commented on scripts/system/libraries/entitySelectionTool.js:
```diff
@@ -796,26 +796,14 @@ SelectionDisplay = (function() {
         borderSize: 1.4
```
@dback2 the changes in this file are pretty simple, basically replace the bounding box lines and fix the draw in front picking.

SamGondelman commented on scripts/system/libraries/WebTablet.js:
```diff
@@ -449,24 +427,6 @@ WebTablet.prototype.calculateWorldAttitudeRelativeToCamera = function (windowPos
-    if (overlayID === this.homeButtonID) {
-        Overlays.editOverlay(this.homeButtonHighlightID, { alpha: 0.0 });
-    }
-};
```
I have no idea why were were handling this in two places.

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16852/HighFidelity-Beta-PR14782-9aae9d1.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16852/HighFidelity-Beta-Interface-PR14782-9aae9d1.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16853/HighFidelity-Beta-PR14782-eb3a53b.apk

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16853/HighFidelity-Beta-PR14782-eb3a53b.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16853/HighFidelity-Beta-Interface-PR14782-eb3a53b.exe

conklin94122 commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16860/HighFidelity-Beta-PR14782-eb3a53b.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16860/HighFidelity-Beta-Interface-PR14782-eb3a53b.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16860/HighFidelity-Beta-PR14782-eb3a53b.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16860/HighFidelity-Beta-Interface-PR14782-eb3a53b.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16861/HighFidelity-Beta-PR14782-7fa676a.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16861/HighFidelity-Beta-PR14782-7fa676a.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16861/HighFidelity-Beta-Interface-PR14782-7fa676a.dmg

SamGondelman commented:
build this please

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16862/HighFidelity-Beta-PR14782-7fa676a.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16862/HighFidelity-Beta-Interface-PR14782-7fa676a.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16862/HighFidelity-Beta-PR14782-7fa676a.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16862/HighFidelity-Beta-Interface-PR14782-7fa676a.exe

qlimnavarro commented:
Stopped Testing will continue tomorrow

qaAdamSmith82 commented:
Build this please

conklin94122 commented:
Labeled with DO NOT MERGE just to make sure it's not accidentally merged into master before we cut the 79.0 release branch.

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16896/HighFidelity-Beta-PR14782-7fa676a.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16896/HighFidelity-Beta-PR14782-7fa676a.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16896/HighFidelity-Beta-Interface-PR14782-7fa676a.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16896/HighFidelity-Beta-PR14782-7fa676a.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16896/HighFidelity-Beta-Interface-PR14782-7fa676a.exe

dback2 commented on scripts/system/libraries/entitySelectionTool.js:
```diff
@@ -1048,7 +1013,19 @@ SelectionDisplay = (function() {
+                return value == selectionBox || value == iconSelectionBox;
+            });
+            if (overlayIncludesNonLayered.length > 0) {
+                intersectObj = Overlays.findRayIntersection(queryRay, true, [], overlayExcludes);
```
Why is overlayIncludesNonLayered not passed in?

dback2 commented on scripts/system/libraries/entitySelectionTool.js:
```diff
@@ -1048,7 +1013,19 @@ SelectionDisplay = (function() {
         }
 
-        var intersectObj = Overlays.findRayIntersection(queryRay, true, overlayIncludes, overlayExcludes);
+        var overlayIncludesLayered = overlayIncludes.filter(function (value, index, arr) {
```
Can you add a comment that this section is to test actual handle intersects before selection box intersects.

dback2 replied:
Would naming like overlayIncludesHandles/overlayIncludesSelectionBoxes be better or is there a specific reason for layered?

dback2 replied:
But actually I'm starting to think maybe having the separate lists prebuilt would be better as they will always be the same?

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/16911/HighFidelity-Beta-PR14782-7d47bfa.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16911/HighFidelity-Beta-PR14782-7d47bfa.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16911/HighFidelity-Beta-Interface-PR14782-7d47bfa.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16911/HighFidelity-Beta-PR14782-7d47bfa.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16911/HighFidelity-Beta-Interface-PR14782-7d47bfa.exe

qaAdamSmith82 commented:
Currently Testing

qaAdamSmith82 commented:
Thus far I have noticed two discrepancies with Nitpick tests across all platforms. 

Tests 6-10 of  **engine / render / effect / highlight / coverage**
![highlight](https://user-images.githubusercontent.com/41648433/52095829-71154280-2579-11e9-9c24-c9e9537b147d.jpg)

And every test under **engine / interaction / pointer / laser /** Every test fails in the same way, I have attached some screenshots that show were the rendering failure is.  I have talked with Sam about these and they may be by design.  However here they are:
Actual Image
![actual image 1](https://user-images.githubusercontent.com/41648433/52096025-2cd67200-257a-11e9-8176-63481aeaa56b.png)
-The sphere at the top of the laser appears partially rendered
-The laser itself looks blurry
-Where the laser intersects with the entity at the bottom looks like there may be some rendering issues
Expected
![expected image](https://user-images.githubusercontent.com/41648433/52096089-6909d280-257a-11e9-8a14-e286d0c697ab.png)
Actual Image
![actual image2](https://user-images.githubusercontent.com/41648433/52096112-80e15680-257a-11e9-989b-035ea2e41725.png)
Expected Image
![expected image 2](https://user-images.githubusercontent.com/41648433/52096115-89399180-257a-11e9-95f7-6e2b0862ccf8.png)

These two potential failures do not repro on master

[hifi-log_14782.zip](https://github.com/highfidelity/hifi/files/2819652/hifi-log_14782.zip)

qaAdamSmith82 commented:
Notification overlays do not disappear:
Repro:
1. Nav to your localhost
2. Have someone on your connections list send out a blast 
3. Observe overlay does not disappear

![notification overlay](https://user-images.githubusercontent.com/41648433/52141640-322fcd00-260b-11e9-8fb2-4aa0296fb72b.jpg)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2822811/hifi-log.txt)

qaAdamSmith82 commented:
The Tablet Becomes transparent loses partial rendering when placed over the HMD friendly logs window.
(This bug is only view-able through HMD mode, and does not show on the screen preview so I could not get a screenshot)
Repro:
1. Nav to your local host
2. Enter HMD mode
3. Bring up the HMD friendly log
4. Move your tablet over it
5. Observe results
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2822892/hifi-log.txt)

qaAdamSmith82 commented:
The blue highlight around the tablet does not disappear after you close the tablet.
(This bug is only view-able through HMD mode, and does not show on the screen preview so I could not get a screenshot)
Repro:
1. Nav to your localhost
2. Bring up the Tablet
3. Position your hand over the tablet so the blue highlight around the tablets borders appears
4. Keep your hand exactly there,  and close the tablet
5. Observe the blue border around the tablet remains on the screen
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2822940/hifi-log.txt)

qaAdamSmith82 commented:
The tablet can get stuck to users hand.
![giphy](https://user-images.githubusercontent.com/41648433/52143260-dd428580-260f-11e9-9e69-4dbc9dfedad1.gif)
This happened the first time I brought up the tablet in HMD mode.
Repro:
1. Nav to your localhost
2. Bring up the tablet
3. Observe it becomes stuck to users hand
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2822946/hifi-log.txt)

qaAdamSmith82 commented:
Crossing lasers causes an artifact in one of the crossed beams
Repro:
1. Nav to your local host
2. Cross hand lasers
3. Observe results

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2823007/hifi-log.txt)
![crossed_beams](https://user-images.githubusercontent.com/41648433/52144027-eaf90a80-2611-11e9-9ec9-635945f6540a.jpg)

qaAdamSmith82 commented:
Lasering over a 2D overlay in cuts through the overlay
Repro:
1. Bring up a 2D overlay (HMD friendly logs for example)
2. Laser over it
3. Observe results
![img_20190201_110439](https://user-images.githubusercontent.com/41648433/52144099-285d9800-2612-11e9-9db7-6a62b266302b.jpg)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2823018/hifi-log.txt)

qaAdamSmith82 commented:
3D Keyboard is broken
Repro:
1. Nav to your localhost in HMD mode
2. Bring up GOTO
3. Observe 3D keyboard is broken, and displays all A's
![img_20190201_110452](https://user-images.githubusercontent.com/41648433/52144334-d23d2480-2612-11e9-835e-6f3c4c50e0b9.jpg)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2823034/hifi-log.txt)

qaAdamSmith82 commented:
The toolbar frequently remains on screen when switching from Desktop to HMD mode.  This screenshot also demonstrates the hand lasers cutting through overlays.
Repro:
1. Nav to your localhost in desktop mode
2. Switch to HMD mode
3. Observe desktop toolbar remains
Repro rate 2/5
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2823044/hifi-log.txt)
![toolbar](https://user-images.githubusercontent.com/41648433/52144535-745d0c80-2613-11e9-9d00-a56dba67af34.jpg)

qaAdamSmith82 commented:
Create is broken. The user cannot edit entities.  They cannot be click dragged to different sizes, they are not select able in HMD mode with hands, or with lasers, and editing the attributes in create properties does not work.
Repro:
1. Nav to your localhost
2. Create a Cube
3. Attempt to laser the center square to resize it
4. Attempt to laser and change the dimensions of the entity on the various points 
5. Attempt to change the size of the entity by entering a different number in "Scale" under entities.
6. Observe no changes are made to the entity.
![no_scale](https://user-images.githubusercontent.com/41648433/52146869-e6385480-2619-11e9-9ea4-e8d7a29c3719.jpg)

qaAdamSmith82 commented:
When the create menu is open the the debug output is so high that the log files reach max size 15 times in one minute.
Repro

1. Nav to  your localhost
2. Open create
3. Open your logs folder
4. Watch the size of your log file zip up to max, and a new one is created.

[hifi-log_2019-02-01_11.42.25_e63cc7cc-cd89-4549-8445-8107ca7180ad.txt](https://github.com/highfidelity/hifi/files/2823194/hifi-log_2019-02-01_11.42.25_e63cc7cc-cd89-4549-8445-8107ca7180ad.txt)
![image](https://user-images.githubusercontent.com/41648433/52147206-d0775f00-261a-11e9-8269-1e67ae9d44a2.png)

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16995/HighFidelity-Beta-PR14782-50bfe84.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/16995/HighFidelity-Beta-Interface-PR14782-50bfe84.dmg

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/16995/HighFidelity-Beta-Interface-PR14782-50bfe84.exe

SamGondelman commented:
Updates:
- Updated nitpick images: https://github.com/highfidelity/hifi_tests/pull/335
- laser and highlight tests should be fixed now
- laser rendering artifacts should be fixed

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17002/HighFidelity-Beta-PR14782-c6080d0.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17002/HighFidelity-Beta-PR14782-c6080d0.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17002/HighFidelity-Beta-Interface-PR14782-c6080d0.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17002/HighFidelity-Beta-PR14782-c6080d0.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17002/HighFidelity-Beta-Interface-PR14782-c6080d0.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17006/HighFidelity-Beta-PR14782-391cca7.apk

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17006/HighFidelity-Beta-Interface-PR14782-391cca7.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17006/HighFidelity-Beta-PR14782-391cca7.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17006/HighFidelity-Beta-Interface-PR14782-391cca7.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17010/HighFidelity-Beta-PR14782-4b406bf.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17010/HighFidelity-Beta-PR14782-4b406bf.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17010/HighFidelity-Beta-Interface-PR14782-4b406bf.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17010/HighFidelity-Beta-PR14782-4b406bf.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17010/HighFidelity-Beta-Interface-PR14782-4b406bf.exe

qaAdamSmith82 commented:
Currently Testing

qaAdamSmith82 commented:
The overlay on mirrors does not work.
Repro:
1. Nav to your localhost
2. Approach the mirror next to the avatar station  in FTUE
3. Observe results

![hifi-snap-by-adamisawake-on-2019-02-04_11-36-25](https://user-images.githubusercontent.com/41648433/52232653-ee8acc80-2871-11e9-868c-def0a084f333.jpg)

[hifi-log_2019-02-04_11.35.55.zip](https://github.com/highfidelity/hifi/files/2829580/hifi-log_2019-02-04_11.35.55.zip)

qaAdamSmith82 commented:
Pausing Testing

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17023/HighFidelity-Beta-PR14782-934abb3.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17023/HighFidelity-Beta-PR14782-934abb3.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17023/HighFidelity-Beta-Interface-PR14782-934abb3.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17023/HighFidelity-Beta-PR14782-934abb3.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17023/HighFidelity-Beta-Interface-PR14782-934abb3.exe

qaAdamSmith82 commented:
Resuming Testing

qaAdamSmith82 commented:
Pausing Testing

qaAdamSmith82 commented:
Resuming Testing

qaAdamSmith82 commented:
Create is still broken, a few new things happen now:
1. Even with the default color set to blue, spheres and cubes appear as brown/black when they are spawned from the create menu. (when the user takes focus off the cube, it will turn blue)
2. Sphere's are spawned as cubes, and when they are resized remain cubes, when focus is taken off the entity they return to a sphere shape
3. Text Entities no longer work
4. I can't figure this one out, but while in my local host, selecting any of the builders_grids in create causes a big brown/black box to appear over the grid. 
5. When entities are created they are always in the foreground, obstructing view of the users avatar.
6. Build hooks (the rings around a created entity) do not scale with the entity if it is enlarged and eventually become hidden by the entity itself.


![sphere_cube](https://user-images.githubusercontent.com/41648433/52308199-988d5600-2951-11e9-9bf5-b8beefabb806.jpg)
![text_entities](https://user-images.githubusercontent.com/41648433/52308211-9cb97380-2951-11e9-85f4-e4e29a087c19.jpg)
![default_color_builder_hooks](https://user-images.githubusercontent.com/41648433/52308217-9fb46400-2951-11e9-99ab-c62338d70e53.jpg)
![builders_grid_create](https://user-images.githubusercontent.com/41648433/52308222-a347eb00-2951-11e9-8f1d-aef7b22e41ff.jpg)
![build_hooks_don t_scale](https://user-images.githubusercontent.com/41648433/52308233-a642db80-2951-11e9-8eb5-3bc30ea0cf78.jpg)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2834114/hifi-log.txt)

qaAdamSmith82 commented:
Mirrors are still not working
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2834171/hifi-log.txt)
[hifi-log_2019-02-05_14.35.36_62cc18ea-70b4-42a1-813e-0586219f6222.txt](https://github.com/highfidelity/hifi/files/2834172/hifi-log_2019-02-05_14.35.36_62cc18ea-70b4-42a1-813e-0586219f6222.txt)
![hifi-snap-by-adamisawake-on-2019-02-05_14-35-34](https://user-images.githubusercontent.com/41648433/52309079-e60ac280-2953-11e9-87e6-2f4b2c894e81.jpg)

qaAdamSmith82 commented:
Lasers still appear to have artifacts in their models:
Repro:
1.  Enter HMD mode
2. Start to use your hand lasers
3. Use your hand lasers over something with a bright backdrop
4. Observe results

[hifi-log_2019-02-05_15.04.42_143da797-6757-47e2-8a49-e3d917d2a0d5.txt](https://github.com/highfidelity/hifi/files/2834269/hifi-log_2019-02-05_15.04.42_143da797-6757-47e2-8a49-e3d917d2a0d5.txt)
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2834270/hifi-log.txt)
![snapshot - 2](https://user-images.githubusercontent.com/41648433/52311232-2588dd00-295b-11e9-98f3-ae7eac70ef5f.png)
![snapshot - 3](https://user-images.githubusercontent.com/41648433/52311235-291c6400-295b-11e9-8d09-39ce874e37d9.png)

qaAdamSmith82 commented:
Lasers double when the user strafes back and forth while flying. The lasers sliding off the hands of the user repro's in master, but the double lasers does not.  This bug is also much more apparent when actually in HMD mode
[hifi-log_2019-02-05_16.09.03.txt](https://github.com/highfidelity/hifi/files/2834334/hifi-log_2019-02-05_16.09.03.txt)

Repro:
1. Enter HMD mode
2. Fly up into the air a ways
3. Trigger both of your hand lasers and aim into the distance
4. Begin strafing back and forth
5. Observe results
![laser](https://user-images.githubusercontent.com/41648433/52312684-47389300-2960-11e9-8174-a3aad7508d4e.gif)

qaAdamSmith82 commented:
Pausing Testing

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17061/HighFidelity-Beta-PR14782-bc11d80.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17061/HighFidelity-Beta-Interface-PR14782-bc11d80.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17061/HighFidelity-Beta-PR14782-bc11d80.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17061/HighFidelity-Beta-Interface-PR14782-bc11d80.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17103/HighFidelity-Beta-PR14782-a665664.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17103/HighFidelity-Beta-PR14782-a665664.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17103/HighFidelity-Beta-Interface-PR14782-a665664.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17103/HighFidelity-Beta-PR14782-a665664.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17103/HighFidelity-Beta-Interface-PR14782-a665664.exe

SamGondelman commented:
- create: fixed
- mirror: possibly fixed, need to check the build from jenkins, but I got negative dimensions working so I'm pretty confident
- laser artifacts: fixed
- double lasers: fixed
- keyboard handle: it's a different color because it's lit.  this won't be fixed now
- The desktop mode toolbar is still hanging around after switching to HMD -> I need to look at this tomorrow

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17115/HighFidelity-Beta-PR14782-e74ff18.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17115/HighFidelity-Beta-PR14782-e74ff18.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17115/HighFidelity-Beta-Interface-PR14782-e74ff18.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17115/HighFidelity-Beta-PR14782-e74ff18.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17115/HighFidelity-Beta-Interface-PR14782-e74ff18.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17136/HighFidelity-Beta-PR14782-c217f8d.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17136/HighFidelity-Beta-PR14782-c217f8d.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17136/HighFidelity-Beta-Interface-PR14782-c217f8d.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17136/HighFidelity-Beta-PR14782-c217f8d.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17136/HighFidelity-Beta-Interface-PR14782-c217f8d.exe

qaAdamSmith82 commented:
* create: fixed (verified)
* mirror: fixed(verified)
* laser artifacts: fixed(verified)
* double lasers: fixed(verified)

There is a new tablet/create bug:
In HMD mode clicking create breaks the 2D overlay on the tablet.
Repro:
1. Nav to your localhost, enter HMD mode
2. Bring up the tablet and select create
3. Observe results
Note - once the overlay is broken by selecting create, it remains broken on other tablet features until the user restarts interface.
![untitled](https://user-images.githubusercontent.com/41648433/52503689-1b552180-2b9b-11e9-8713-b140375bc473.jpg)
![untitled2](https://user-images.githubusercontent.com/41648433/52503700-2019d580-2b9b-11e9-891b-e779e2e50926.jpg)


[hifi-log.txt](https://github.com/highfidelity/hifi/files/2846697/hifi-log.txt)

qaAdamSmith82 commented:
The mini-tablet appears broken.  With use the mini tablet set to on in settings, I have been unable to summon it through the normal hand motions.  I tested it against master and was able to summon the mini tablet there.
Repro:
1.  Nav to your local host
2. Enter HMD mode
3. Make sure the mini tablet is turned on in settings
4. Attempt to summon the mini tablet
5. Observe results

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2847361/hifi-log.txt)
![mt](https://user-images.githubusercontent.com/41648433/52513679-3fc3f480-2bc1-11e9-8668-98492be8bd9a.jpg)

qaAdamSmith82 commented:
The 3D keyboard loses focus when the user quickly enters text if lasers instead of mallets are used.
Repro:
1. Nav to your local host
2. Enter HMD mode 
3. Bring up your tablet and select settings>general
4. Change the 3D keyboard input from mallets to lasers
5. Bring up GOTO 
6. Enter some random text, quickly
7. Observe results
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2847405/hifi-log.txt)

qaAdamSmith82 commented:
Nitpick results from highfidelity/hifi_tests#335:
https://hifi-qa.s3.amazonaws.com/TestResults--2019-02-08_18-42-29(local)[DESKTOP-IQCUEB4]/TestResults.html
100% Pass

The one test that looks like a possible fail is a false positive due to timing
content / entity / model / renderLayer / front

When run by itself it looks correct:
![tests content entity model renderlayer front 00000](https://user-images.githubusercontent.com/41648433/52515744-73f4e080-2bd4-11e9-882d-67001f425d62.png)

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17188/HighFidelity-Beta-PR14782-b76c5b3.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17188/HighFidelity-Beta-PR14782-b76c5b3.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17188/HighFidelity-Beta-Interface-PR14782-b76c5b3.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17188/HighFidelity-Beta-PR14782-b76c5b3.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17188/HighFidelity-Beta-Interface-PR14782-b76c5b3.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17194/HighFidelity-Beta-PR14782-f53ccf7.apk

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17196/HighFidelity-Beta-PR14782-96f32a9.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17196/HighFidelity-Beta-PR14782-96f32a9.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17196/HighFidelity-Beta-Interface-PR14782-96f32a9.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17196/HighFidelity-Beta-PR14782-96f32a9.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17196/HighFidelity-Beta-Interface-PR14782-96f32a9.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17214/HighFidelity-Beta-PR14782-5e044ba.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17214/HighFidelity-Beta-PR14782-5e044ba.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17214/HighFidelity-Beta-Interface-PR14782-5e044ba.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17214/HighFidelity-Beta-PR14782-5e044ba.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17214/HighFidelity-Beta-Interface-PR14782-5e044ba.exe

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17230/HighFidelity-Beta-PR14782-b418e43.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17230/HighFidelity-Beta-PR14782-b418e43.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17230/HighFidelity-Beta-Interface-PR14782-b418e43.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17230/HighFidelity-Beta-PR14782-b418e43.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17230/HighFidelity-Beta-Interface-PR14782-b418e43.exe

AndrewMeadows commented on libraries/entities-renderer/src/RenderablePolyLineEntityItem.cpp:
```diff
@@ -220,11 +234,17 @@ void PolyLineEntityRenderer::updateGeometry() {
+                binormal = glm::normalize(glm::cross(tangent, normal));
+
+                // Check to make sure binormal is not a NAN. If it is, don't add to vertices vector
+                if (binormal.x != binormal.x) {
```
Maybe use `glm::isnan(binormal)`?

AndrewMeadows commented on libraries/entities-renderer/src/RenderableWebEntityItem.cpp:
```diff
@@ -115,35 +115,41 @@ bool WebEntityRenderer::needsRenderUpdateFromTypedEntity(const TypedEntityPointe
+        return false;
+    })) {
         return true;
     }
```
Maybe just:
```
return resultWithReadLock<bool>([&] {
...
});
```

SamGondelman replied:
This is following a pattern in other similar methods and is intended to make it easier to add additional checks in the future.  I'll fix everything else.

AndrewMeadows commented on libraries/entities/src/EntityItemProperties.cpp:
```diff
@@ -1927,7 +1927,7 @@ QScriptValue EntityItemProperties::copyToScriptValue(QScriptEngine* engine, bool
     }
 
-    if (!psuedoPropertyFlagsActive || psueudoPropertyFlags.test(EntityPsuedoPropertyFlag::FaceCamera)) {
+    if ((!psuedoPropertyFlagsActive || psueudoPropertyFlags.test(EntityPsuedoPropertyFlag::FaceCamera)) && _type != EntityTypes::PolyLine) {
```
Swap the order of `if (A && B)` to be `if (B && A)` for faster typical evaluation.

AndrewMeadows commented on libraries/entities/src/EntityTreeElement.cpp:
```diff
@@ -139,16 +139,22 @@ bool EntityTreeElement::bestFitBounds(const glm::vec3& minPoint, const glm::vec3
+    // We only check the collidable filters for non-local entities, because local entities are always collisionless
+    bool collidable = !entity->getCollisionless() && (entity->getShapeType() != SHAPE_TYPE_NONE);
+    if (hostType != entity::HostType::LOCAL) {
+        if ((!searchFilter.doesPickCollidable() && collidable) || (!searchFilter.doesPickNonCollidable() && !collidable)) {
```
Not a blocker: for slightly faster evaluation swap the order of checks here:
`if ((!A && B) || (!C && !D))`  --> `if ((B && !A) || (!D && !C))`

AndrewMeadows commented on libraries/entities/src/WebEntityItem.cpp:
```diff
@@ -306,6 +303,18 @@ WebInputMode WebEntityItem::getInputMode() const {
+    return resultWithReadLock<bool>([&] {
+        return _showKeyboardFocusHighlight;
+    });
+}
```
Reading and writing to `bool` in these cases are `atomic` operations and don't require a lock, so you can't be trying to prevent these methods from being messed up by others... you must be trying to prevent these methods from messing up other, larger operations elsewhere.

qaAdamSmith82 commented:
The 3D Keyboard still loses focus when using lasers for text entry.
Repro:
1. Nav to your localhost
2. Change input from mallets to lasers
3. Begin entering text in GOTO
4. Press back/delete :<:   (click it a few times)
5. Begin typing again
6. Observe results
7. Being entering text in GOTO again
8. Close the 3D keyboard but leave GOTO up
9. Select the text field in GOTO so the 3D keyboard reappears
10. Begin entering text
11. Observe results

Repro rate 4/5

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2862455/hifi-log.txt)

qaAdamSmith82 commented:
Clicking Menu in the tablet causes the X (close) button at the bottom of the tab to turn white/lose its overlay.
Repro:
1. Bring up the tablet in HMD mode
2. Click Menu
3. Observe results:
Repro rate 5/5

[hifi-log.txt](https://github.com/highfidelity/hifi/files/2862500/hifi-log.txt)

qaAdamSmith82 commented:
Interface crashes when the user selects 'log out' while in HMD mode from the tablet.
Repro:
1. Launch PR14782
2. Sign into High Fidelity
3. Nav to your local host
4. Enter HMD mode
5. Bring up the tablet and select 'log out'
6. Observe results
[hifi-log.txt](https://github.com/highfidelity/hifi/files/2862619/hifi-log.txt)
[hifi-log_2019-02-13_15.31.27.txt](https://github.com/highfidelity/hifi/files/2862620/hifi-log_2019-02-13_15.31.27.txt)

hifi-gustavo commented:
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/17264/HighFidelity-Beta-PR14782-3f8bfc5.apk

hifi-gustavo commented:
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17264/HighFidelity-Beta-PR14782-3f8bfc5.dmg

hifi-gustavo commented:
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/17264/HighFidelity-Beta-Interface-PR14782-3f8bfc5.dmg

hifi-gustavo commented:
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17264/HighFidelity-Beta-PR14782-3f8bfc5.exe

hifi-gustavo commented:
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/17264/HighFidelity-Beta-Interface-PR14782-3f8bfc5.exe
